src/Screens/VotingPage.jsx

```diff
@@ -0,0 +1,135 @@
+import { useEffect, useRef, useState } from 'react';
+import { useNavigate } from 'react-router-dom';
+import { getPartyCandidates, getParties, voteForCandidate, voteForParty, voteBlank } from '../API/VotingAPI';
+import './VotingPage.css';
+
+function VotingPage() {
+  const navigate = useNavigate();
+  const [parties, setParties] = useState([]);
+  const [partyCandidates, setPartyCandidates] = useState({});
+  const [selectedParty, setSelectedParty] = useState(null);
+  const [selectedCandidate, setSelectedCandidate] = useState(null);
+  const [isTextLarge, setIsTextLarge] = useState(false); // New state variable
+  const hasFetchedData = useRef(false);
+
+  const fetchParties = async () => {
+    const parties = await getParties();
+    return parties;
+  };
+
+  const fetchPartyCandidates = async (partyId) => {
+    const candidates = await getPartyCandidates(partyId);
+    return candidates;
+  };
+
+  useEffect(() => {
+    const fetchData = async () => {
+      const fetchedParties = await fetchParties();
+      setParties(fetchedParties);
+
+      const candidatesData = {};
+      for (const party of fetchedParties) {
+        const candidates = await fetchPartyCandidates(party._id);
+        candidatesData[party._id] = candidates;
+      }
+      setPartyCandidates(candidatesData);
+      hasFetchedData.current = true;
+    };
+
+    fetchData();
+  }, []);
+
+  const handleSelectParty = (event) => {
+    const partyId = event.target.value;
+    setSelectedParty(partyId === selectedParty ? null : partyId);
+    setSelectedCandidate(null);
+  };
+
+  const handleSelectCandidate = (event) => {
+    const candidateId = event.target.value;
+    setSelectedCandidate(candidateId === selectedCandidate ? null : candidateId);
+    setSelectedParty(null);
+  };
+
+  const handleCastVote = async () => {
+    if (selectedParty) {
+      await voteForParty(selectedParty);
+    } else if (selectedCandidate) {
+      await voteForCandidate(selectedCandidate);
+    } else {
+      voteBlank();
+    }
+    setSelectedParty(null);
+    setSelectedCandidate(null);
+    sessionStorage.setItem('verified', 'false');
+    navigate('/login');
+  };
+
+  return (
+    <>
+      <div className="header">
+        <h1>Folketingsvalget 20xx</h1>
+        <div className="upper-right-text">Opstillingskreds</div>
+      </div>
+      <button className="text-size-button" onClick={() => setIsTextLarge(!isTextLarge)}>
+        {isTextLarge ? 'Normal tekststørrelse' : 'Forstør tekst'}
+      </button>
+      <h3 className={`${isTextLarge ? 'large-text' : ''}`}>Sæt X i rubrikken til venstre for et partinavn eller et kandidatnavn.</h3>
+      <p className={`${isTextLarge ? 'large-text' : ''}`}>Du kan kun sætte ét X på stemmesedlen.</p>
+      {parties.length === 0 && <p>Indlæser...</p>}
+      {parties.map((party) => (
```

Comment:
I believe this might cause issues since there is no check for whether `parties` is actually an array (and not e.g. `null` or `undefined`)

Reply:
This is addressed in 480e409

Reply:
It has not been solved. I believe you have changed the wrong line (perhaps accidentally?).

Line 80 should probably be something along the lines of:
```js
{parties?.length && parties.map((party) => (
```
Or perhaps using Array.isArray could be an alternative.

Reply:
ok fr this time no cap d146f3f